plugins/axios-as-datasource/src/index.ts

```diff
@@ -21,6 +23,89 @@ class DataSource extends GSDataSource {
     try {
       const client = this.client as AxiosInstance;
 
+      if (args.retry) {
+        const { max_attempts, type, interval, min_interval, max_interval } = args.retry;
+
+        axiosRetry(client, {
+          retries: max_attempts,
+          retryDelay:(
+            retryNumber: number,
+            error: AxiosError<any, any>,
+          )=> {
+            switch (type) {
+              case 'constant':
+                logger.info(`Attempt ${retryNumber}: Retrying request with ${type} retry delay. Error: ${error.message}`);
+                return interval;
+      
+              case 'random':
+                let min = Math.ceil(min_interval);
+                let max = Math.floor(max_interval);
+                logger.info(`Attempt ${retryNumber}: Retrying request with ${type} retry delay. Error: ${error.message}`);
+
+                return Math.floor(Math.random() * (max + 1 - min) + min);
+      
+              case 'exponential':
+                const delay = 2 ** retryNumber * interval;
+                const randomSum = delay * 0.2 * Math.random();
+                logger.info(`Attempt ${retryNumber}: Retrying request with ${type} retry delay. Error: ${error.message}`);
+                return delay + randomSum;
+            }
+            return 0;
+          },
+          retryCondition: (error) => {
```

Comment:
does this have to be a function or can we just set it to 
`retryCondition:true`

Reply:
retryCondition is a callback to further control if a request should be retried. default: it retries if it is a network error or a 5xx error on an idempotent request (GET, HEAD, OPTIONS, PUT or DELETE).

Setting retryCondition: true means that the request will always be retried, regardless of the error. If you want more control over when to retry requests, you should customize the retryCondition function based on the error received. For example, you might want to retry only on certain status codes or under specific error conditions. If you don't have specific conditions to check and want to retry on any error, you can leave it as is (return true).